gcloud-java-core/src/main/java/com/google/gcloud/PageImpl.java

```diff
@@ -38,6 +40,38 @@
     Page<T> nextPage();
   }
 
+  /**
+   * Base class for page fetchers.
+   *
+   * @param <T> the value type that the page holds. {@code T} should implement
+   *     {@code java.io.Serializable}.
+   * @param <ServiceOptionsT> the service options type.
+   */
+  @SuppressWarnings("rawtypes")
+  public abstract static class BasePageFetcher<T, ServiceOptionsT extends ServiceOptions>
```

Comment:
We should keep naming consistent, that is using either `T` and `O` or `ItemT` (just guessing) and `ServiceOptionsT`.